bundles/org.openhab.binding.melcloud/src/main/java/org/openhab/binding/melcloud/internal/handler/MelCloudDeviceHandler.java

```diff
@@ -0,0 +1,310 @@
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.melcloud.internal.api.json.DeviceStatus;
+import org.openhab.binding.melcloud.internal.config.AcDeviceConfig;
+import org.openhab.binding.melcloud.internal.exceptions.MelCloudCommException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link MelCloudDeviceHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Luca Calcaterra - Initial contribution
+ * @author Pauli Anttila - Refactoring
+ */
+
+public class MelCloudDeviceHandler extends BaseThingHandler {
+
+    private static final int EFFECTIVE_FLAG_POWER = 0x01;
+    private static final int EFFECTIVE_FLAG_OPERATION_MODE = 0x02;
+    private static final int EFFECTIVE_FLAG_TEMPERATURE = 0x04;
+    private static final int EFFECTIVE_FLAG_FAN_SPEED = 0x08;
+    private static final int EFFECTIVE_FLAG_VANE_VERTICAL = 0x10;
+    private static final int EFFECTIVE_FLAG_VANE_HORIZONTAL = 0x100;
+
+    private final Logger logger = LoggerFactory.getLogger(MelCloudDeviceHandler.class);
+    private AcDeviceConfig config;
+    private MelCloudAccountHandler melCloudHandler;
+    private DeviceStatus deviceStatus;
+    private ScheduledFuture<?> refreshTask;
+
+    public MelCloudDeviceHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing {} handler.", getThing().getThingTypeUID());
+
+        Bridge bridge = getBridge();
+        if (bridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Bridge Not set");
+            return;
+        }
+
+        config = getThing().getConfiguration().as(AcDeviceConfig.class);
+        logger.debug("A.C. device config: {}", config);
+
+        initializeBridge(bridge.getHandler(), bridge.getStatus());
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Running dispose()");
+        if (refreshTask != null) {
+            refreshTask.cancel(true);
+            refreshTask = null;
+        }
+        melCloudHandler = null;
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        logger.debug("bridgeStatusChanged {} for thing {}", bridgeStatusInfo, getThing().getUID());
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            initializeBridge(bridge.getHandler(), bridgeStatusInfo.getStatus());
+        }
+
+        if (bridgeStatusInfo.getStatus() == ThingStatus.ONLINE) {
+            startAutomaticRefresh();
```

Comment:
I added the call in `initializeBridge` because the binding returned "no initial data" and this "fixed" it.
Issue is discussed here:
https://github.com/lucacalcaterra/org.openhab.binding.melcloud/issues/9
I wasn't to happy about this as I didn't know the actual source of the issue, it was on my list for a later time to check this further.
Would you like to have this validated before merging this PR?

Reply:
I just tested again after by removing the additional call I added (in `initializeBridge`) and now I cannot reproduce the "no initial data" issue anymore.
I'll test it some more the coming days.

Reply:
What is the status of:

> I just tested again after by removing the additional call I added (in initializeBridge) and now I cannot reproduce the "no initial data" issue anymore.
I'll test it some more the coming days.

Reply:
Thanks for pointing this out, at the time I couldn't reproduce with existing config but I did some more tests today and it happens when adding Things without the Bridge changing state.
I made a change so this gets called in `initializeBridge` and removed the call in `bridgeStatusChanged`.
I can now no longer reproduce the issue.